### ISSUE TITLE
Update UI for editing shortcuts

### ISSUE THREAD

**Issue by fallaciousreasoning** [brave/brave-browser#29349]
@aguscruiz has provided a nice design for the Shortcuts UI which should live on the settings page - we should update the UI to match this:

https://www.figma.com/file/tLXWGCpNoiJxDZDdpfordj/Desktop-Settings?node-id=3617-77370&t=HKTZrtO87yiUZkol-0
![image](https://user-images.githubusercontent.com/7678024/228096620-18544971-a21e-47bc-8e09-9153c61118f8.png)

Blocks https://github.com/brave/brave-browser/issues/354

**Comment by stephendonner** [comment]
Note: we'll want https://github.com/brave/brave-browser/issues/30757 to have landed before testing this issue 👍

**Comment by LaurenWags** [comment]
Marking as `QA/Blocked` until we have this officially landed in 1.53.x

**Comment by MadhaviSeelam** [comment]
Verification `PASSED` using 
```
Brave | 1.53.102 Chromium: 114.0.5735.133 (Official Build) (64-bit)
-- | --
Revision | fbfa2ce68d01b2201d8c667c2e73f648a61c4f4a-refs/branch-heads/5735@{#1270}
OS | Windows 11 Version 22H2 (Build 22621.1848)
```

Install 1.53.102
launch Brave
opened brave://flags
set brave://flags/#brave-commands to Enabled
clicked Relaunch
opened brave://settings/system/shortcuts


![image](https://github.com/brave/brave-browser/assets/98358127/4d5c0b3f-dcf8-4cfc-a689-1978261f2def)

Shortcuts: 

*Back* - Delete `AltGr Arrowleft` shortcut`

- Hovered over the shortcut
- clicked (-) key 
- successfully deleted `AltGr  Arrowleft` shortcut since I don't have the AltGr key on my keyboard 
- clicked Reset to reinstate the shortcut 


example | example | example
---- | -----  | -----
![2023-06-22_13h27_31](https://github.com/brave/brave-browser/assets/98358127/2e80afcb-2353-4b91-bb95-cae6a397e11f)|![2023-06-22_13h27_55](https://github.com/brave/brave-browser/assets/98358127/71035841-7b0c-45fb-80e6-48e6ca95ef3b)|![2023-06-22_13h34_43](https://github.com/brave/brave-browser/assets/98358127/31da0335-a5b3-4e6f-9a62-bd3fef83d73f)


*Front* - Override the shortcut  message

- Click `Add` button
- `Create a new shortcut` dialog opened as expected
- press `Alt` key on the keyboard
- `This combination is being used for ``Focus menu bar".  Saving will override that shortcut` message is shown

example | example
---- | -----
![image](https://github.com/brave/brave-browser/assets/98358127/47b578fc-4c44-4c77-91c6-262600cc386b)|![2023-06-22_13h25_27](https://github.com/brave/brave-browser/assets/98358127/8c288d9b-d73e-4ba2-a542-31c89af98ff2)

*Bookmark tab* - override the shortcut with another key

- Click `Add` button
- press CTRL+ ALT+B on the keyboard
- Click Save on the dialog
- Bookmark tab shortcut now shows `Control Alt B`
- Saved bookmark by using CTRL+ ALT+B

example | example |example | example
---- | ----- | ---- | -----
![2023-06-22_15h03_45](https://github.com/brave/brave-browser/assets/98358127/ba10df42-6d18-4753-8232-aae544eff978)|![2023-06-22_15h04_06](https://github.com/brave/brave-browser/assets/98358127/e616dfb5-f635-4f2b-a590-2f3d0077e23a)|![2023-06-22_15h04_18](https://github.com/brave/brave-browser/assets/98358127/f487de5c-742d-4011-86f2-75b855763711)|![2023-06-22_15h07_06](https://github.com/brave/brave-browser/assets/98358127/37ea7f08-26f6-4d8b-9cce-4e0458b578b1)


*Stop* - Add  a new key

- Click `Add` button
- press S key on the keyboard
- clicked `Save` button

example | example
---- | -----
![image](https://github.com/brave/brave-browser/assets/98358127/e0bc5780-63af-4bc1-ae91-81bfdef8c1d1)|![image](https://github.com/brave/brave-browser/assets/98358127/acf03df0-413a-4a61-82a4-9d9c05c4e47e)|![image](https://github.com/brave/brave-browser/assets/98358127/2eb271f9-86a4-4640-9d0f-ea3b826ef359)

Search a shortcut

- Search by Duplicate
- confirmed `Duplicate tab` shortcut is shown 

![image](https://github.com/brave/brave-browser/assets/98358127/fde8e8f3-5501-48d0-8908-3448ffaba04f)
